deprecation/__init__.pyi

```diff
@@ -0,0 +1,33 @@
+"""Stubs file for `deprecation.py`.
+
+Created using mypy's `stubgen`, types manually corrected.
+"""
+
+import datetime
+from packaging import version
+from typing import Callable, Optional, Union
+
+message_location: str
+
+class DeprecatedWarning(DeprecationWarning):
+    function: str
+    deprecated_in: Union[datetime.date, version.Version, None]
+    removed_in: Union[datetime.date, version.Version, None]
+    details: Union[datetime.date, version.Version, None]
```

Comment:
The examples in the readme pass `str` for `deprecated_in`, `removed_in`, and `details`. I don't think `details` is ever a date/version.

Reply:
It looks like [the code](https://github.com/briancurtin/deprecation/blob/master/deprecation.py#L197) supports `datetime.date`.

Reply:
Yes, `removed_in` can be a date. I don't see any indication that the other two fields can be a date.

Based on the code and docstrings, it looks like:
- `deprecated_in`: version, str, or None
- `removed_in`: date, version, str, or None
- `details`: str or None